### ISSUE TITLE
How to get attention score? "return_attn_probs=True" is not work.

### ISSUE THREAD

**Issue by UnableToUseGit** [Dao-AILab/flash-attention#1357]
version:  flash-attn 2.7.0.post2

When I use the function "flash_attn_func" and I set return_attn_probs=True, it not returns a correct S_dmask

code:
```
attn_output, softmax_lse, S_dmask = flash_attn_func(query_states, key_states, value_states, dropout, softmax_scale=softmax_scale, causal=causal, return_attn_probs=True)
print(S_dmask)
>> tensor([], device='cuda:0', dtype=torch.float16)
```

**Comment by tridao** [comment]
As the docstring says, it's for testing only.
```
        return_attn_probs: bool. Whether to return the attention probabilities. This option is for
           testing only. The returned probabilities are not guaranteed to be correct
           (they might not have the right scaling).
```
In fact for dropout=0.0 we will return None.

**Comment by UknowSth** [comment]
Could you help me understand why the number of tokens in `S_dmask` is different from that in `softmax_lse`? When I ran a self-attention inference, I noticed that the number of tokens in S_dmask was 32 more than in softmax_lse.

```
S_dmask.shape:torch.Size([2, 24, 4992, 4992])
softmax_lse.shape:torch.Size([24, 4960])
```

**Comment by tridao** [comment]
Again `S_dmask` is not meant to be used other than testing.  You can see how the tests uses it here:
https://github.com/Dao-AILab/flash-attention/blob/22c0358f4ba7999a15dbe27989ce163e5edb5693/tests/test_flash_attn.py#L613

Tbh if you want the attn probabilities you should just compute it in pytorch.
